src/router_core/route_control.c

```diff
@@ -396,7 +408,18 @@ void qdr_route_connection_closed_CT(qdr_core_t *core, qdr_connection_t *conn)
             al = DEQ_NEXT_N(REF, al);
         }
 
-        cid->open_connection = 0;
+        //
+        // Remove our own entry in the connection list
+        //
+        qdr_connection_ref_t * cref = DEQ_HEAD(cid->connection_refs);
+        while (cref) {
+            if (cref->conn == conn) {
+                DEQ_REMOVE(cid->connection_refs, cref);
```

Comment:
I believe this will leak one qdr_connection_ref_t each time it is run.  Why did you not use qdr_del_connection_ref?